ext/storage_sources/azure_store/azure_storage_source.cpp

```diff
@@ -335,17 +380,30 @@ azure_file_system_terminate(WT_FILE_SYSTEM *file_system, WT_SESSION *session)
     return 0;
 }
 
+// Checks if a file exists in the container.
 static int
-azure_file_exists(WT_FILE_SYSTEM *file_system, WT_SESSION *session, const char *name, bool *existp)
+azure_file_system_exists(
+  WT_FILE_SYSTEM *file_system, WT_SESSION *session, const char *name, bool *existp)
 {
     WT_UNUSED(file_system);
     WT_UNUSED(session);
-    WT_UNUSED(name);
-    WT_UNUSED(existp);
+    azure_file_system *azure_fs = reinterpret_cast<azure_file_system *>(file_system);
 
+    std::cout << "azure_file_system_exists: Checking object: " << name << " exists in Azure."
+              << std::endl;
+
+    // Check whether the object exists in the cloud.
+    azure_fs->azure_conn->object_exists(name, *existp);
+    if (!*existp) {
```

Comment:
What's the motivation behind making this an error? It doesn't seem unreasonable to ask if a file exists and receive "no" as a response. I would've thought that returning non-zero values is for when we fail to check existence at all (e.g. we've lost our network connection).